### ISSUE TITLE
Luarocks does not properly support Github https://github.com/... URLs.

### ISSUE THREAD

**Issue by spc476** [luarocks/luarocks#1408]
I have a bunch of Luarocks where the source location starts with a git: URL pointing to Gitlab.  Gitlab no longer support git: URLs, only SSH: or HTTPS, per https://github.blog/2021-09-01-improving-git-protocol-security-github/ .  I tried changing a few rocks to using https: but they fail to work properly.  The rocks in question are org.conman.dns and org.conman.uuid.  

Platform: Linux
LuaRocks: 3.5.0
Lua: 5.4

To reproduce: attempt to install org.conman.dns or org.conman.uuid (with the https: source location) or org.conman.env (with the older git: source location).

**Comment by spc476** [comment]
Nevermind---I read the documentation and found my answer.

**Comment by tst2005** [comment]
Yes, LuaRocks 3.8.0 is the answer (because https://github.com/luarocks/luarocks/commit/9ff512e35455939f02eaec2318e3acc77782fdeb )

**Comment by tuxd3v** [comment]
So, what is the answer for this?
I understand that weneed to update luarocks to LuaRocks 3.8.0,
But how will I call my source url??
I have this:
``lua
source = {
	url	= "git://github.com/tuxd3v/lnewt",
	branch	= "master"
}
``

**Comment by tuxd3v** [comment]
Ok, so for what it seems it replace git by git+https.
But does I need extra packages on the system to deal with https?
I am on debian bullseye
for example the luarocks package doesn't even install git, it also doesn't have lua-sec as a dependency, but for what I understand it will need lua-sec right?

**Comment by hishamhm** [comment]
@tuxd3v the existing `git://github.com` URLs continue to work in LuaRocks 3.8.0. In general, to use `git+https` you can use `git+https://` URLs but it's not needed to change existing rockspecs that refer to GitHub.

To use Git one needs to have `git` installed. For `https://` URLs, LuaRocks can either use LuaSocket + LuaSec, or also either `curl` or `wget`. It should autodetect whichever one you have installed.

Any packaging issues related to dependencies should be brought up with the distribution maintainers.
